packages/design-system/src/components/Button/Button.js

```diff
@@ -26,19 +45,21 @@ const Button = React.forwardRef((props, ref) => {
     ariaExpanded,
     className,
     type: htmlType,
+    kind,
+    iconPosition,
     ...buttonProps
   } = props;
 
   const isDisabled = disabled || loading;
   const type = submit ? 'submit' : htmlType || 'button';
-  let buttonType = null;
+  const isValidKind = kind && acceptedKinds.includes(kind);
+  const deprecatedKind = getDeprecatedKind(primary, destructive, secondary);
+  const buttonKind = (isValidKind && kind) || deprecatedKind || null;
 
-  if (primary) {
-    buttonType = 'primary';
-  } else if (destructive) {
-    buttonType = 'destructive';
-  } else if (secondary) {
-    buttonType = 'secondary';
+  if (deprecatedKind) {
+    noticeAboutDeprecation(
+      `deprecated prop '${deprecatedKind}' in Button component - please use 'kind' prop instead`
```

Comment:
I think we should inform the AA teams about this change before merging it.

Reply:
@kamilmateusiak good idea - could you take care of that? I'm not a member of AA private channels and I guess it's easier to announce there changes like this 😅